src/Neo/Builders/WitnessBuilder.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// WitnessBuilder.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Network.P2P.Payloads;
+using Neo.VM;
+using System;
+
+namespace Neo.Builders
+{
+    public sealed class WitnessBuilder
+    {
+        private byte[] _invocationScript = [];
+        private byte[] _verificationScript = [];
+
+        private WitnessBuilder() { }
+
+        public static WitnessBuilder CreateEmpty()
+        {
+            return new WitnessBuilder();
+        }
+
+        public WitnessBuilder AddInvocation(Action<ScriptBuilder> config)
+        {
+            var sb = new ScriptBuilder();
+            config(sb);
+            _invocationScript = sb.ToArray();
+            return this;
+        }
+
+        public WitnessBuilder AddInvocation(byte[] bytes)
+        {
+            _invocationScript = bytes;
+            return this;
+        }
+
+        public WitnessBuilder AddVerification(Action<ScriptBuilder> config)
+        {
+            var sb = new ScriptBuilder();
+            config(sb);
+            _verificationScript = sb.ToArray();
+            return this;
+        }
+
+        public WitnessBuilder AddVerification(byte[] bytes)
+        {
+            _verificationScript = bytes;
+            return this;
+        }
```

Comment:
If it's already defined, it's remplaced, this doesn't happend with `AddWitness` for example

Reply:
because you can have more than one witness.

Reply:
`WitnessBuilder` is the builder for a single witness. `AddWitness` is adding multiple witnesses.

Reply:
Throw an error?

Reply:
builders normally don't do that. But I can.

Reply:
Done!